trinity/sync/full/state.py

```diff
@@ -253,11 +230,42 @@ def deregister_peer(self, peer: BasePeer) -> None:
                 return
 
             candidates = list(not_yet_requested.difference(self._peer_missing_nodes[peer]))
-            batch = candidates[:eth_constants.MAX_STATE_FETCH]
+            batch = tuple(candidates[:eth_constants.MAX_STATE_FETCH])
             not_yet_requested = not_yet_requested.difference(batch)
             self.request_tracker.active_requests[peer] = (time.time(), batch)
-            self.logger.debug("Requesting %d trie nodes to %s", len(batch), peer)
-            peer.sub_proto.send_get_node_data(batch)
+            asyncio.ensure_future(self._request_and_process_nodes(peer, batch))
+
+    async def _request_and_process_nodes(self, peer: ETHPeer, batch: Tuple[Hash32, ...]) -> None:
+        self.logger.debug("Requesting %d trie nodes from %s", len(batch), peer)
+        nodes = await peer.requests.get_node_data(batch)
+
+        self.logger.debug("Got %d NodeData entries from %s", len(nodes), peer)
```

Comment:
Should we also include here how many nodes were originally requested? I know it's in the previous log but there will usually be lots of noise between them so it'd be good to have that info here as well.